src/internal/group_api/mod.rs

```diff
@@ -526,6 +530,82 @@ pub async fn group_create<CR: rand::CryptoRng + rand::RngCore>(
     resp.try_into()
 }
 
+#[derive(Debug, Clone)]
+pub struct GroupUpdatePrivateKeyResult {
+    needs_rotation: bool,
+}
+
+impl GroupUpdatePrivateKeyResult {
+    /// True if this group's private key requires additional rotation
+    pub fn needs_rotation(&self) -> bool {
+        self.needs_rotation
+    }
+}
+
+/// Rotate the group's private key. The public key for the group remains unchanged.
+pub async fn group_rotate_private_key<CR: rand::CryptoRng + rand::RngCore>(
+    recrypt: &Recrypt<Sha256, Ed25519, RandomBytes<CR>>,
+    group_id: &GroupId,
+    auth: &RequestAuth,
+    device_private_key: &PrivateKey,
+) -> Result<GroupUpdatePrivateKeyResult, IronOxideErr> {
+    let group_info = group_get_request(&auth, group_id).await?;
+    let admins = match group_info.admin_ids {
```

Comment:
Would you be willing to explore making the vecs in the request structure just have UserIds in them?